src/field_type.h

```diff
@@ -23,15 +24,37 @@ class JsonObject;
 enum phase_id : int;
 enum body_part : int;
 
-struct field_effect_data {
+struct field_effect {
     efftype_id id;
-    time_duration min_duration;
-    time_duration max_duration;
-    int intensity;
-    body_part bp;
-    bool inside_immune;
+    time_duration min_duration = 0_seconds;
+    time_duration max_duration = 0_seconds;
+    int intensity = 0;
+    body_part bp = num_bp;
+    bool is_environmental = false;
+    bool inside_immune = false;
+    bool outside_immune = false;
+    int inside_chance = 0;
+    int outside_chance = 0;
+    int field_intensity_delta = 0;
+    game_message_type env_message_type = m_neutral;
+    translation env_message;
+    translation env_message_npc;
+    time_duration get_duration() const {
+        return rng( min_duration, max_duration );
+    }
+    std::string get_env_message() const {
+        return env_message.translated();
+    }
+    std::string get_env_message_npc() const {
+        return env_message_npc.translated();
+    }
+    effect get_effect( const time_point &start_time = calendar::turn ) const {
+        return effect( &id.obj(), get_duration(), bp, false, intensity, start_time );
+    }
 };
 
+using field_effects_collection = std::vector<field_effect>;
```

Comment:
This typedef is used only 3 times, it would be easier to read if you just used `std::vector<field_effect>` directly at those points.